cp_core/src/task/transform/common.rs

```diff
@@ -9,33 +11,31 @@ use crate::{
     util::error::{CpError, CpResult},
 };
 
-use super::config::RootTransformConfig;
-
 /// Base transform trait. Takes
 pub trait Transform {
     fn run(&self, main: LazyFrame, ctx: Arc<DefaultPipelineContext>) -> CpResult<LazyFrame>;
 }
 
-pub trait SubTransformConfig {
+pub trait TransformConfig {
     fn validate(&self) -> Vec<CpError>;
-    fn transform(self) -> Box<dyn Transform>;
+    fn transform(&self) -> Box<dyn Transform>;
 }
 
-/// The root transform node that runs all its substages
+/// The root transform node that runs all its subtransforms
 pub struct RootTransform {
     label: String,
     input: String,
     output: String,
-    stages: Vec<Box<dyn Transform>>,
+    subtransforms: Vec<Box<dyn Transform>>,
```

Comment:
This is OK, do not change (keep RootTransform.subtransforms, change to RootTransformConfig.steps)